### PR TITLE
Announce 4.145

### PR DESCRIPTION
**Pull request by Atry**

### CODE REVIEW

File: _posts/2022-01-21-hhvm-4.145.markdown [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: "HHVM 4.144"
```
Comment (fredemmott) [inline _posts/2022-01-21-hhvm-4.145.markdown:2]: Wrong version

File: _posts/2022-01-21-hhvm-4.145.markdown [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+layout: post
+author: atry
+category: blog
+---
+
+HHVM 4.146 is released! This release marks the end of support of 4.137;
```
Comment (fredemmott) [inline _posts/2022-01-21-hhvm-4.145.markdown:8]: different wrong version :)

File: _posts/2022-01-21-hhvm-4.145.markdown [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+4.138, 4.139, 4.140, 4.143, 4.144 remain supported, as do the 4.102 and 4.128 LTS releases.
+4.141 and 4.142 were cancelled due to the holiday season.
+
+
+# Highlights
+
+- Support throw on dynamic, e.g. `function foo(dynamic $d): void { throw $d; }`
+- Add support for attributes on class constants
+- Strictify several built-in functions
+- Several bug fixes
```
Comment (fredemmott) [inline _posts/2022-01-21-hhvm-4.145.markdown:18]: Remove or expand; this isn't useful by itself

File: _posts/2022-01-21-hhvm-4.145.markdown [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+# Breaking Changes
+
+- Hack will raise 4068 on `?mixed` and similar if they're in a type hint position. For example, `f<?mixed>()` or `new C<?mixed>()` now results a typing error.
```
Comment (fredemmott) [inline _posts/2022-01-21-hhvm-4.145.markdown:22]: add what 4068 is

Comment (Atry) [reply _posts/2022-01-21-hhvm-4.145.markdown]: Updated to "the typing error 4068"

File: _posts/2022-01-21-hhvm-4.145.markdown (hunk 3 above)
Comment (Atry) [reply _posts/2022-01-21-hhvm-4.145.markdown]: Removed

File: _posts/2022-01-21-hhvm-4.145.markdown (hunk 1 above)
Comment (Atry) [reply _posts/2022-01-21-hhvm-4.145.markdown]: Updated

File: _posts/2022-01-21-hhvm-4.145.markdown (hunk 2 above)
Comment (Atry) [reply _posts/2022-01-21-hhvm-4.145.markdown]: Updated

**Comment by Atry** [comment]
Still waiting for the completion of `promote-nightly-to-release`

**Review by fredemmott — APPROVED** [review]

File: _posts/2022-01-21-hhvm-4.145.markdown [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+HHVM 4.145 is released! This release marks the end of support of 4.137;
+4.138, 4.139, 4.140, 4.143, 4.144 remain supported, as do the 4.102 and 4.128 LTS releases.
+4.141 and 4.142 were cancelled due to the holiday season.
+
+
+# Highlights
+
+- Support throw on dynamic, e.g. `function foo(dynamic $d): void { throw $d; }`
+- Add support for attributes on class constants
+- Strictify several built-in functions
```
Comment (fredemmott) [inline _posts/2022-01-21-hhvm-4.145.markdown:17]: Is this "built-in functions now have more accurate type signatures" or "they are implemented in hack strict/with fewer fixmes, but no user-visible changes"?

File: _posts/2022-01-21-hhvm-4.145.markdown [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+HHVM 4.145 is released! This release marks the end of support of 4.137;
+4.138, 4.139, 4.140, 4.143, 4.144 remain supported, as do the 4.102 and 4.128 LTS releases.
+4.141 and 4.142 were cancelled due to the holiday season.
+
+
+# Highlights
+
+- Support throw on dynamic, e.g. `function foo(dynamic $d): void { throw $d; }`
+- Add support for attributes on class constants
```
Comment (lexidor) [inline _posts/2022-01-21-hhvm-4.145.markdown:16]: I would like to elaborate on this list item. The [reflection hhi](https://github.com/facebook/hhvm/blob/master/hphp/hack/hhi/reflection.hhi) does not contain a type for class constants (only type constants).

Class constant attributes must inherit from the new `ClassConstantAttribute` interface. It is currently not possible to use runtime reflection on a class constant. These attributes can (only) be parsed by static analysis tools such as hhast.

Comment (Atry) [reply _posts/2022-01-21-hhvm-4.145.markdown]: Updated. Thank you!

File: _posts/2022-01-21-hhvm-4.145.markdown (hunk 5 above)
Comment (Atry) [reply _posts/2022-01-21-hhvm-4.145.markdown]: Move the signature change about `curl_multi_exec`  to Breaking Changes

**Pull request merged**